srcdeps-core/src/main/java/org/l2x6/srcdeps/core/config/ScmRepository.java

```diff
@@ -94,6 +101,65 @@ public Builder urls(List<String> urls) {
 
     }
 
+    /** The period character that delimits the segments of {@link #id} values */
+    public static final char ID_DELIMITER = '.';
+
+    /**
+     * Checks that the given {@code id} is a valid id. If no violation is found the given {@code id} is returned.
+     * Otherwise an {@link IllegalArgumentException} is thrown.
+     *
+     * Valid IDs are much like Java packages: they are sequences of Java identifiers concatenad by {@code '.'}
+     * character.
+     *
+     * @param id
+     *            the ID to check
+     * @return the {@code id} passed to this method
+     * @throws IllegalArgumentException
+     *             is the given {@code id} violates some of the requirements
+     */
+    public static String assertValidId(String id) {
+        if (id == null) {
+            throw new IllegalArgumentException(
+                    String.format("Invalid %s.id [%s]: cannot be null", ScmRepository.class.getSimpleName(), id));
```

Comment:
It is very minor thing, but having error message patterns as a private static final fields would probably help keeping visual structure of the code clear, for ex.:
if (id == null) 
  throw new IllegalArgumentException(String.format(ID_NULL, id));
if (id.isEmpty()) 
  throw new IllegalArgumentException(String.format(ID_EMPTY, id));
...